pyscf/nao/gw.py

```diff
@@ -116,7 +116,7 @@ def get_wmin_wmax_tmax_ia_def(self, tol):
     E_fermi = self.fermi_energy
     E_homo = amax(E[where(E<=E_fermi)])
     #first issue >=,E gap goes to 0 for H, when HOMO and E_fermi are identical, and 'tmax_def' goes to infinity
-    E_gap  = amin(E[where(E>E_fermi)]) - E_homo  
+    E_gap  = amin(E[where(E>=E_fermi)]) - E_homo  
```

Comment:
Why >= instead of > ?

Reply:
earlier it was >= and I changed and committed for Hydrogen case it to prevent going to infinity when 
wmin_def = sqrt(tol * (d**3) * (E_gap**3)/(d**2+E_gap**2))
however, I found a similar situation for F by considering spin=3. Again we have a problem with nfermi when it shows 0.5.